### PR TITLE
Add `PollyLexicon` module

### PR DESCRIPTION
**Pull request by danarbaugh**
This adds [Polly](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/polly/index.html) support by removing the only resource the service manages outside of sound files output to local or S3; [Lexicons](https://docs.aws.amazon.com/polly/latest/dg/gs-put-lexicon.html).

# Testing

Polly Lexicons were created using the Setup Bash Script included below. AWS Nuke then showed this successful removal within its logs with `PollyLexicon` included in the configuration.

```
us-east-1 - PollyLexicon - [Alphabet: "ipa", LanguageCode: "en-US", LastModified: "2023-07-11 15:44:41.906 +0000 UTC", LexemesCount: "2", LexiconArn: "arn:aws:polly:us-east-1:123456789012:lexicon/testlexicon", Name: "testlexicon", Size: "555"] - triggered remove
...
us-east-1 - PollyLexicon - [Alphabet: "ipa", LanguageCode: "en-US", LastModified: "2023-07-11 15:44:41.906 +0000 UTC", LexemesCount: "2", LexiconArn: "arn:aws:polly:us-east-1:123456789012:lexicon/testlexicon", Name: "testlexicon", Size: "555"] - waiting
...
us-east-1 - PollyLexicon - [Alphabet: "ipa", LanguageCode: "en-US", LastModified: "2023-07-11 15:44:41.906 +0000 UTC", LexemesCount: "2", LexiconArn: "arn:aws:polly:us-east-1:123456789012:lexicon/testlexicon", Name: "testlexicon", Size: "555"] - removed
```

# Setup Bash Script

```
#!/bin/bash

set -ex

# Specify the name and content of the lexicon
LEXICON_NAME="testlexicon"
cat <<EOF > lexicon.xml
<?xml version="1.0" encoding="UTF-8"?>
<lexicon version="1.0" xmlns="http://www.w3.org/2005/01/pronunciation-lexicon"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:schemaLocation="http://www.w3.org/2005/01/pronunciation-lexicon
    http://www.w3.org/TR/2007/CR-pronunciation-lexicon-20071212/pls.xsd"
    alphabet="ipa"
    xml:lang="en-US">
    <lexeme>
        <grapheme>h3ll0</grapheme>
        <alias>hello</alias>
    </lexeme>
    <lexeme>
        <grapheme>w0rld</grapheme>
        <alias>world</alias>
    </lexeme>
</lexicon>
EOF

# Create the lexicon
echo "Creating the lexicon..."
aws polly put-lexicon \
  --name $LEXICON_NAME \
  --content file://lexicon.xml

echo "Lexicon created with name: $LEXICON_NAME"

# Get information about the lexicon
echo "Getting information about the lexicon..."
aws polly get-lexicon \
  --name $LEXICON_NAME
```

### CODE REVIEW

File: resources/polly-lexicons.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	register("PollyLexicons", ListPollyLexicons)
+}
+
+func ListPollyLexicons(sess *session.Session) ([]Resource, error) {
+	svc := polly.New(sess)
+	resources := []Resource{}
+
+	listLexiconsInput := &polly.ListLexiconsInput{}
+
+	listOutput, err := svc.ListLexicons(listLexiconsInput)
```
Comment (bjoernhaeuser) [inline resources/polly-lexicons.go:25]: Do we need paging here?

**Comment by der-eismann** [comment]
@danarbaugh would you be so kind to implement our suggestions?

File: resources/polly-lexicons.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	_, err := lexicon.svc.DeleteLexicon(deleteInput)
+	return err
+}
+
+func (lexicon *PollyLexicon) Properties() types.Properties {
+	properties := types.NewProperties()
+	properties.Set("Name", lexicon.name)
+	properties.Set("Alphabet", lexicon.attributes.Alphabet)
+	properties.Set("LanguageCode", lexicon.attributes.LanguageCode)
+	properties.Set("LastModified", lexicon.attributes.LastModified)
```
Comment (der-eismann) [inline resources/polly-lexicons.go:65]:
```suggestion
	properties.Set("LastModified", lexicon.attributes.LastModified.Format(time.RFC3339))
```

Could you please add this? Time needs to be formatted correctly for the filters to work

**Review by der-eismann — COMMENTED** [review]
Just one tiny change for the time property, rest looks good :+1:

**Comment by ekristen** [comment]
This will be implemented via https://github.com/ekristen/aws-nuke/pull/339

If you have a chance, please check it out and let us know if you run into an issues by opening an issue over on the fork.

----

Please see the copy of the notice from the README about the deprecation of this project. Sven was kind enough to grant me access to help triage and close issues and pull requests that have already been addressed in the actively maintained fork. Some additional information is located in the [welcome issue](https://github.com/ekristen/aws-nuke/issues/276) for more information. 

> [!CAUTION]
> This repository for aws-nuke is no longer being actively maintained. We recommend users to switch to the actively maintained fork of this project at [ekristen/aws-nuke](https://github.com/ekristen/aws-nuke).
> We appreciate all the support and contributions we've received throughout the life of this project. We believe that the fork will continue to provide the functionality and support that you have come to expect from aws-nuke.
> Please note that this deprecation means we will not be addressing issues, accepting pull requests, or making future releases from this repository.
> Thank you for your understanding and support.

**Pull request closed without merging**